zio/ndjsonio/reader.go

```diff
@@ -128,5 +128,14 @@ again:
 		return nil, fmt.Errorf("line %d: %w", r.scanner.Stats.Lines, err)
 	}
 	outType := r.zctx.LookupTypeRecord(zv.Type.(*zng.TypeRecord).Columns)
-	return zng.NewRecordCheck(outType, 0, zv.Bytes)
+	record, err := zng.NewRecordCheck(outType, 0, zv.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	ts, err := record.AccessTime("ts")
+	if err == nil {
+		record.Ts = ts
+	}
```

Comment:
Use zng.NewRecord to avoid a map access by taking advantage of TypeRecord.TsCol.
```suggestion
	record, err := zng.NewRecord(outType, zv.Bytes)
	if err != nil {
		return nil, err
	}
	if err := record.TypeCheck(); err != nil {
		return nil, err
	}
```

Reply:
👍